File: src/February.js
function February(que,poi,va,out,line){
var query=que
    var point=poi
    var vari=va
    var output=out
    var i=line

    
        if(query[0][2]==1){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1); 
            }else{
            vari[point]=[0,(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]
            }
        }else if(query[0][2]==2){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1); 
            }else{
            vari[point]=[1,(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]
            }
        }else if(query[0][2]==3){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000<0){
                console.error("calcore_error:"+(i+1))
                console.error("変数ポインタが0未満になってしまいます")
                process.exit(1); 
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1); 
            }else{
                point=(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000
            }
        }else if(query[0][2]==4){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1); 
            }else if((point+(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)<0){
                console.error("calcore_error:"+(i+1))
                console.error("変数ポインタが0未満になってしまいます")
                process.exit(1); 
            }else{
                point+=(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000
            }
        }else if(query[0][2]==5){
          //  変数ポインタの移動(ポインタ＝変数)
          if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))    
                console.error("引数がありません")
                process.exit(1);             
          }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1); 
          }else if((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000<0){
                console.error("calcore_error:"+(i+1))
                console.error("指定した変数の場所は0未満の場所です")
                process.exit(1); 
          }else{
          if(vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("指定した変数は値がありません")
                process.exit(1); 
          }
          else if(vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000][1]<0){
                console.error("calcore_error:"+(i+1))
                console.error("変数ポインタが0未満になってしまいます")
                process.exit(1); 
          }else{
          point=vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000][1]
            }
        }
        }else if(query[0][2]==6){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000<0){
                console.error("calcore_error:"+(i+1))
                console.error("指定した変数の場所は0未満の場所です")
                process.exit(1); 
            }else{
            vari[point]=vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]
            }
            //引数が日付形式ではなかったら、コピー先が、unifinedになる。
        }else if(query[0][2]==7){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1); 
            }else if((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000<0){
            console.error("calcore_error:"+(i+1))
            console.error("指定した変数の場所は0未満の場所です")
            process.exit(1); 
            }else{
            vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]=[0,point]
            }
        }else if(query[0][2]==8){
            if(vari[point]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("ポインタ先の変数に値がありません。")
                process.exit(1); 
            }else{
                vari[point][0]=1
            }
        }else if(query[0][2]==9){
            if(vari[point]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("ポインタ先の変数に値がありません。")
                process.exit(1); 
            }else{
                vari[point][0]=0
            }
        }else if(query[0][2]==10){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1); 
            }else if((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000<0){
                console.error("calcore_error:"+(i+1))
                console.error("指定した変数の場所は0未満の場所です")
                process.exit(1); 
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1)
            }
            else{
            vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]=vari[point]
            }
        }else if(query[0][2]==11){
            if(query[1]==undefined){
                console.error("calcore_error:"+(i+1))
                console.error("引数がありません")
                process.exit(1)
            }else if(isNaN((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000)){
                console.error("calcore_error:"+(i+1))
                console.error("引数は日付形式以外では指定できません")
                process.exit(1)
            }else if((new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000<0){
                console.error("calcore_error:"+(i+1))
                console.error("指定した変数の場所は0未満の場所です")
                process.exit(1)
            }else if(vari[point]==undefined){
                vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]=[0,0]
            }else{
                vari[(new Date(query[1][0],query[1][1]-1,query[1][2])-new Date(2000, 0, 1))/86400000]=[0,vari[point][0]+1]
            }
        }

    return [point,vari,output,i]
}
module.exports={
    February
}

File: src/python/february.py
from .utils import get_days_from_base, exit_with_error

def february(query, point, vari, output, i):
    sub_cmd = int(query[0][2])
    
    if sub_cmd == 1:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        vari[point] = [0, float(days)]
        
    elif sub_cmd == 2:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        vari[point] = [1, float(days)]
        
    elif sub_cmd == 3:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        if days < 0: exit_with_error(i + 1, "変数ポインタが0未満になってしまいます", "Pointer to variable would underflow.")
        point = days
        
    elif sub_cmd == 4:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        if (point + days) < 0: exit_with_error(i + 1, "変数ポインタが0未満になってしまいます", "Pointer to variable would underflow.")
        point += days
        
    elif sub_cmd == 5:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        if days < 0: exit_with_error(i + 1, "指定した変数の場所は0未満の場所です", "Specified memory address is negative.")
        val = vari.get(days)
        if val is None: exit_with_error(i + 1, "指定した変数は値がありません", "Specified variable is undefined.")
        if val[1] < 0: exit_with_error(i + 1, "変数ポインタが0未満になってしまいます", "Pointer to variable would underflow.")
        point = val[1]
        
    elif sub_cmd == 6:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None or days < 0: exit_with_error(i + 1, "指定した変数の場所は0未満の場所です", "Specified memory address is negative.")
        vari[point] = vari.get(days)
        
    elif sub_cmd == 7:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        if days < 0: exit_with_error(i + 1, "指定した変数の場所は0未満の場所です", "Specified memory address is negative.")
        vari[days] = [0, float(point)]
        
    elif sub_cmd == 8:
        if vari.get(point) is None: exit_with_error(i + 1, "ポインタ先の変数に値がありません。", "Value at pointer is undefined.")
        vari[point][0] = 1
        
    elif sub_cmd == 9:
        if vari.get(point) is None: exit_with_error(i + 1, "ポインタ先の変数に値がありません。", "Value at pointer is undefined.")
        vari[point][0] = 0
        
    elif sub_cmd == 10:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        if days < 0: exit_with_error(i + 1, "指定した変数の場所は0未満の場所です", "Specified memory address is negative.")
        vari[days] = vari.get(point)
        
    elif sub_cmd == 11:
        if len(query) < 2: exit_with_error(i + 1, "引数がありません", "No arguments provided.")
        days = get_days_from_base(query[1][0], query[1][1], query[1][2])
        if days is None: exit_with_error(i + 1, "引数は日付形式以外では指定できません", "Argument must be a valid date.")
        if days < 0: exit_with_error(i + 1, "指定した変数の場所は0未満の場所です", "Specified memory address is negative.")
        if vari.get(point) is None:
            vari[days] = [0, 0.0]
        else:
            vari[days] = [0, float(vari[point][0] + 1)]
            
    return [point, vari, output, i]
